utopia-remix/app/routes/internal.projects.$id.restore.tsx

```diff
@@ -3,9 +3,15 @@ import { Params } from '@remix-run/react'
 import { restoreDeletedProject } from '../models/project.server'
 import { ensure, handle, requireUser } from '../util/api.server'
 import { Status } from '../util/statusCodes'
+import { ALLOW } from '../handlers/validators'
 
 export async function action(args: ActionFunctionArgs) {
-  return handle(args, { POST: handleRestoreDeletedProject })
+  return handle(args, {
+    POST: {
+      handler: handleRestoreDeletedProject,
+      validator: ALLOW,
```

Comment:
this should also validate the project access

Reply:
fixed